### PR TITLE
Added loans and returns pages

### PR DESCRIPTION
**Pull request by benjimarshall**
Some screenshots:

### New book details page
bookish_book_details2

### After checking a book out
bookish_checked_out

### After returning a book
bookish_returned

### CODE REVIEW

File: Bookish.DataAccess/BookishService.cs [hunk 1]
```diff
@@ -37,6 +41,24 @@ public IEnumerable<Book> GetBooks()
         {
             return GetCatalogue().FirstOrDefault(book => book.Isbn == isbn);
         }
```
Comment (FridaTveit) [inline Bookish.DataAccess/BookishService.cs:43]: Empty line between methods please

File: Bookish.DataAccess/BookishService.cs [hunk 2]
```diff
@@ -126,5 +150,28 @@ public void AddBook(Book book, int numberOfCopies)
                 numberOfCopies
             });
         }
+
+        public void CheckoutBook(LoanedBook book, string userId)
+        {
+            var dueDate = DateTime.Now.AddDays(14);
```
Comment (FridaTveit) [inline Bookish.DataAccess/BookishService.cs:156]: I would extract 14 into a suitably named constant 🙂

File: Bookish.DataAccess/BookishService.cs [hunk 3]
```diff
@@ -126,5 +150,28 @@ public void AddBook(Book book, int numberOfCopies)
...
+                userId,
+                dueDate
+            });
+        }
+
+        public void ReturnBook(LoanedBook book)
+        {
+            var sqlString = @"DELETE FROM loans WHERE bookId = @copyId;";
+
+            connection.Execute(sqlString, new { copyId = book.CopyId });
```
Comment (FridaTveit) [inline Bookish.DataAccess/BookishService.cs:174]: For the modifying methods here which don't return anything, it's a good idea to check if they succeeded (by checking how many rows were modified which execute should return) and log a warning if they didn't 🙂

File: Bookish.Web/Controllers/HomeController.cs [hunk 4]
```diff
@@ -85,6 +87,89 @@ public IActionResult BookAdded(string isbn)
...
         }
 
+        public IActionResult CheckoutBook()
+        {
+            return StatusCode(404);
+        }
+
+        [HttpPost]
+        [ActionName("CheckoutBook")]
+        public IActionResult CheckoutBookPost(string copyId)
```
Comment (FridaTveit) [inline Bookish.Web/Controllers/HomeController.cs:97]: Is there any reason not to name this just `CheckoutBook`? 🙂

File: Bookish.Web/Controllers/HomeController.cs [hunk 5]
```diff
@@ -85,6 +87,89 @@ public IActionResult BookAdded(string isbn)
...
+        }
+
+        public IActionResult ReturnBook()
+        {
+            return StatusCode(404);
+        }
+
+        [HttpPost]
+        [ActionName("ReturnBook")]
+        public IActionResult ReturnBookPost(string copyId)
```
Comment (FridaTveit) [inline Bookish.Web/Controllers/HomeController.cs:123]: Same question as above about naming it `ReturnBook` 🙂

**Review by FridaTveit — COMMENTED** [review]
A few comments, otherwise looks good 🙂

File: Bookish.DataAccess/BookishService.cs (hunk 1 above)
Comment (benjimarshall) [reply Bookish.DataAccess/BookishService.cs:44]: Yep, sorry! Fixed in cd3d310 🙂

File: Bookish.DataAccess/BookishService.cs (hunk 2 above)
Comment (benjimarshall) [reply Bookish.DataAccess/BookishService.cs]: Yes, sounds good! Done in cd3d310 🙂

File: Bookish.DataAccess/BookishService.cs (hunk 3 above)
Comment (benjimarshall) [reply Bookish.DataAccess/BookishService.cs]: Yes that's a good idea. I haven't set up a logging system for this task so I've just written the error messages to the console. Done in cd3d310 🙂

File: Bookish.Web/Controllers/HomeController.cs (hunk 4 above)
Comment (benjimarshall) [reply Bookish.Web/Controllers/HomeController.cs]: Nope, changed in cd3d310 🙂

File: Bookish.Web/Controllers/HomeController.cs (hunk 5 above)
Comment (benjimarshall) [reply Bookish.Web/Controllers/HomeController.cs]: Nope, changed in cd3d310 🙂

File: Bookish.DataAccess/BookishService.cs [hunk 6]
```diff
@@ -153,25 +155,37 @@ public void AddBook(Book book, int numberOfCopies)
...
             {
                 bookId = book.CopyId,
                 userId,
                 dueDate
             });
+
+            if (rowsChanged != 1)
+            {
+                Console.WriteLine($"Expected to see one row changed after checking out copy {book.CopyId}, instead " +
+                                  $"{rowsChanged} rows were changed.");
```
Comment (FridaTveit) [inline Bookish.DataAccess/BookishService.cs:174]: User id is probably also useful context here? 🙂

File: Bookish.DataAccess/BookishService.cs [hunk 7]
```diff
@@ -24,6 +24,7 @@ public interface IBookishService
     public class BookishService : IBookishService
     {
         private readonly IDbConnection connection;
+        const int LoanPeriod = 14;
```
Comment (FridaTveit) [inline Bookish.DataAccess/BookishService.cs:27]: Can be private? 🙂

**Review by FridaTveit — COMMENTED** [review]
A couple of minor things, otherwise looks good 🙂 

Another extension I would recommend is adding a proper logger, e.g. [serilog](https://serilog.net/) which is a popular logger we're using on HEE 🙂

File: Bookish.DataAccess/BookishService.cs (hunk 6 above)
Comment (benjimarshall) [reply Bookish.DataAccess/BookishService.cs]: Yep, added in bfc6c5a 🙂

File: Bookish.DataAccess/BookishService.cs (hunk 7 above)
Comment (benjimarshall) [reply Bookish.DataAccess/BookishService.cs]: Yep, done in bfc6c5a 🙂

**Review by FridaTveit — APPROVED** [review]
Looks good 🙂

**Pull request merged**